feature/dialogue-system

Banananaman91 opened the pull request:
Begun creation of dialogue system, currently working on allowing selection responses for customised dialogue trees

Banananaman91 committed:
Begun dialogue system

Begun creation of dialogue system, currently working on allowing selection responses for customised dialogue trees

Banananaman91 committed:
Base functionality finished

Dialogue allows for multiple scriptable objects to be created, and unique dialogue stored. Also has the ability to get npc images including specific moods

RubyNova commented on GameProject/Assets/TurnBasedAssets/Scripts/Dialogue/Dialogue.cs:
```diff
@@ -0,0 +1,39 @@
+    }
+
+    [Serializable]
+    public class Message
```
Move this into another file.

RubyNova commented on GameProject/Assets/TurnBasedAssets/Scripts/Dialogue/Dialogue.cs:
```diff
@@ -0,0 +1,39 @@
+    }
+
+    [Serializable]
+    public class Response
```
Move this into another file.

RubyNova commented on GameProject/Assets/TurnBasedAssets/Scripts/Dialogue/NpcImages.cs:
```diff
@@ -0,0 +1,21 @@
+    }
+    
+    [Serializable]
+    public class NpcMoods
```
Move this into another file.

RubyNova commented on GameProject/Assets/TurnBasedAssets/Scripts/Dialogue/RenderDialogue.cs:
```diff
@@ -0,0 +1,89 @@
+        [SerializeField] private Image _dialogueBackground;
+        [SerializeField] private GameObject _dialogueBox;
+        private List<Button> _responseOptions = new List<Button>();
+        private Image previousImage;
```
This doesn't follow the codebase's naming convention for private fields.

RubyNova commented on GameProject/Assets/TurnBasedAssets/Scripts/Dialogue/RenderDialogue.cs:
```diff
@@ -0,0 +1,89 @@
+                button.GetComponentInChildren<Text>().text = response.Reply;
+                _responseOptions.Add(button);
+                button.onClick.AddListener(() => PlayParagraphCycle(npcMessage, npcImages, response.Next));
+                ++buttonCount;
```
why not just use a `for` loop here instead?

Banananaman91 replied:
Personal preference to use foreach like this, I find it clearer and nicer to use

RubyNova replied:
I advise changing it still. This is precisely what `for` was meant for - not to mention you are introducing additional overhead by relying on `foreach`, albeit a tiny amount, its slightly noisy code.

RubyNova commented on GameProject/Assets/TurnBasedAssets/Scripts/Dialogue/RenderDialogue.cs:
```diff
@@ -0,0 +1,89 @@
+
+        private void EndDialogue()
+        {
+            _pageName.text = "";
```
please prefer `string.Empty` over `""` as intent is much clearer.

RubyNova commented on GameProject/Assets/TurnBasedAssets/Scripts/Dialogue/RenderDialogue.cs:
```diff
@@ -0,0 +1,89 @@
+        private void EndDialogue()
+        {
+            _pageName.text = "";
+            _pageText.text = "";
```
please prefer `string.Empty` over `""` as intent is much clearer.

RubyNova commented on GameProject/Assets/TurnBasedAssets/Scripts/Dialogue/Dialogue.cs:
```diff
@@ -0,0 +1,39 @@
+namespace TurnBasedAssets.Scripts.Dialogue
+{
+    [CreateAssetMenu(fileName = "Dialogue", menuName = "Dialogue", order = 1)]
+    [Serializable]
```
Please combine these attributes into a single set of square brackets.

RubyNova commented on GameProject/Assets/TurnBasedAssets/Scripts/Dialogue/RenderDialogue.cs:
```diff
@@ -0,0 +1,89 @@
+
+        public void PlayParagraphCycle(Dialogue npcDialogue, NpcImages npcImages, int paragraphNumber)
+        {
+            if (_dialogueBox.activeSelf == false) _dialogueBox.SetActive(true);
```
prefer a NOT operator over a comparison to `false`.

Banananaman91 committed:
Improvements

Improvements to code, including separating classes, in response to code review feedback

Banananaman91 committed:
Changed foreach loop to for loop

Banananaman91 committed:
Changed mood id for string text

Mood checking now takes text input, name mood image files accordingly and input that mood when creating messages

Banananaman91 committed:
Changed image handling to use npc names instead of id

Replaced ID, now on the npc image list you can add the npc name for that mood list, then the npc name given on the dialogue is used to find the correct image list

Banananaman91 committed:
Cleaned up scene and code ready for review

Removed unnecessary scene objects and cleaned up code

RubyNova commented on GameProject/Assets/TurnBasedAssets/Scripts/Dialogue/DialogueObject.cs:
```diff
@@ -0,0 +1,27 @@
+            _pageRender.PlayParagraphCycle(_dialogue, _npcImages,_startMessage);
+        }
+
+        public void DeSelect()
```
Is this supposed to be empty?

Banananaman91 replied:
Yes, there was no functionality to add but the Interface required it to be there

RubyNova commented on GameProject/Assets/TurnBasedAssets/Scripts/Dialogue/RenderDialogue.cs:
```diff
@@ -0,0 +1,127 @@
+                    }
+                }
+
+                if (!_dialogueBox.activeSelf)
```
You can one-line this if statement.

RubyNova commented on GameProject/Assets/TurnBasedAssets/Scripts/Dialogue/RenderDialogue.cs:
```diff
@@ -0,0 +1,127 @@
+                    _dialogueBox.SetActive(true);
+                }
+
+                if (IsPreviousImageNotNull)
```
You can one-line this if statement.

RubyNova commented on GameProject/Assets/TurnBasedAssets/Scripts/Dialogue/RenderDialogue.cs:
```diff
@@ -0,0 +1,127 @@
+
+        public void PlayParagraphCycle(Dialogue npcDialogue, NpcImages npcImages, int paragraphNumber)
+        {
+            if (paragraphNumber < 0)
```
have you considered just `return`ing after calling `EndDialogue()` here? It'd reduce your scope count by one.

Banananaman91 replied:
I don't understand?

RubyNova reviewed (changes requested):
Very minor things. Once these are dealt with I'll approve.

Banananaman91 committed:
Code review changes

Banananaman91 committed:
Finished code review changes

RubyNova commented on GameProject/Assets/TurnBasedAssets/Scripts/Dialogue/DialogueObject.cs:
```diff
@@ -0,0 +1,27 @@
+        [SerializeField] private RenderDialogue _pageRender;
+        [SerializeField] private int _startMessage;
+        
+        public void Select()
```
You can expression body this.

RubyNova commented on GameProject/Assets/TurnBasedAssets/Scripts/Dialogue/RenderDialogue.cs:
```diff
@@ -0,0 +1,122 @@
+                RenderPageText(npc.NpcName, sb.ToString());
+                yield return new WaitForSeconds(_sentenceSpeed);
+            }
+            yield return null;
```
This yield return null doesn't need to be here I don't think?

RubyNova reviewed (changes requested):
Ok just these two tweaks and it should be good to go.

Banananaman91 committed:
Review changes

RubyNova reviewed (approved):
Good to go bar your scene conflict.

Banananaman91 committed:
Merge branch 'Development' into feature/dialogue-system
